MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -131,6 +135,30 @@ open class RouteProgress: NSObject {
         self.legIndex = legIndex
         super.init()
         currentLegProgress = RouteLegProgress(leg: currentLeg, stepIndex: 0, alertLevel: alertLevel)
+        
+        var coordinateIndex = 0
+        for leg in route.legs {
+            
+            var congestionTravelTimesSegmentsByLeg: [[(CongestionLevel, TimeInterval)]] = []
+            
+            for (stepIndex, step) in leg.steps.enumerated() {
+                if let segmentCongestionLevels = leg.segmentCongestionLevels, let expectedSegmentTravelTimes = leg.expectedSegmentTravelTimes  {
+                    
+                    let stepCoordinatesCount = Int(step.coordinateCount) + coordinateIndex - 1
+                    
+                    let congestionSegment = Array(segmentCongestionLevels[coordinateIndex-stepIndex..<stepCoordinatesCount - stepIndex])
```

Comment:
If I understand correctly, `coordinateIndex` is an index into a leg’s array of coordinates, while `stepIndex` is an index into a leg’s array of steps. Why is it correct to subtract one from the other, given that a step may have more than one coordinate?